src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/external_repacking_adjuster.cpp

```diff
@@ -14,59 +14,144 @@
 namespace ov {
 namespace intel_cpu {
 
+const size_t BrgemmExternalRepackingAdjuster::brgemm_kernel_rank = 2;
+
 BrgemmExternalRepackingAdjuster::BrgemmExternalRepackingAdjuster(const ov::snippets::lowered::LinearIRCPtr& linear_ir,
                                                                  const CPURuntimeConfigurator* configurator)
     : snippets::lowered::pass::RuntimeOptimizer(configurator) {
     const auto& params = linear_ir->get_parameters();
     for (size_t i = 0; i < params.size(); ++i) {
         const auto& param = params[i];
-        const auto consumers = param->get_output_port_connector(0)->get_consumers();
-        const bool brgemm_with_extracted_repacking =
-            std::any_of(consumers.begin(), consumers.end(), [](const ov::snippets::lowered::ExpressionPort& port) {
-                auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(port.get_expr()->get_node());
-                return brgemm && brgemm_utils::with_repacking(brgemm->get_type()) && port.get_index() == 1;
-            });
-        if (brgemm_with_extracted_repacking) {
-            m_param_idces_with_external_repacking.insert(i);
-            // Ticket 157339: Support non-planar layout
-            OPENVINO_ASSERT(ov::snippets::utils::is_planar_layout(configurator->get_io_descs()[i]->get_layout()),
-                            "Non-planar layout is not supported for external repacking");
+        const auto& shape_infer_consumers = ov::snippets::utils::get_first_child_shape_infer_expr_seq(param);
+        const auto& out = shape_infer_consumers.empty() ? param->get_output_port(0)
+                                                        : shape_infer_consumers.back()->get_output_port(0);
+        const auto consumers = out.get_connected_ports();
+
+        for (const auto& consumer : consumers) {
+            auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(consumer.get_expr()->get_node());
+            if (brgemm && brgemm_utils::with_repacking(brgemm->get_type()) && consumer.get_index() == 1) {
+                const auto src_prc = brgemm->get_input_element_type(0);
+                const auto wei_prc = brgemm->get_input_element_type(1);
+                const auto isa = brgemm_utils::get_primitive_isa(src_prc, brgemm_utils::with_amx(brgemm->get_type()));
+                const auto inner_n_block = brgemm_utils::repacking::compute_inner_n_block(wei_prc);
+                const auto is_transposed_b =
+                    BrgemmCopyB::is_transposed(m_configurator->get_io_descs()[i]->get_layout());
+                auto config = BrgemmCopyBKernelConfig(src_prc, wei_prc, isa, false, is_transposed_b, inner_n_block);
+                m_executors[i] = std::make_shared<BrgemmCopyBKernelExecutor>(
+                    static_cast<const CPURuntimeConfigurator*>(m_configurator)->get_cache(),
```

Comment:
`static_cast` is generally not safe for downcasting, please use dynamic_cast + nullptr check
BTW, why don't we use `configurator` here? It's already of desired type